app/actions.cpp

```diff
@@ -1290,9 +1290,9 @@ void Modify::delMetadatum(Exiv2::Image* pImage, const ModifyCmd& modifyCmd) {
     }
   }
   if (modifyCmd.metadataId_ == MetadataId::xmp) {
-    Exiv2::XmpData::iterator pos;
     const Exiv2::XmpKey xmpKey(modifyCmd.key_);
-    if ((pos = xmpData.findKey(xmpKey)) != xmpData.end()) {
+    auto pos = xmpData.findKey(xmpKey);
+    if (pos != xmpData.end()) {
```

Comment:
[suggestion] You can still use the C++17 feature to declare the variable inside the if and afterwards do the comparison.

Reply:
I know about it. Doesn't look as nice IMO